soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -46,76 +167,634 @@ Color_RGBA8 Color_Skipped_Main_Default              = { 160, 160, 160, 255 };
+        case SCENE_SHADOW_TEMPLE:
+        case SCENE_BOTTOM_OF_THE_WELL:
+        case SCENE_ICE_CAVERN:
+        case SCENE_GERUDO_TRAINING_GROUND:
+        case SCENE_GANONS_TOWER:
+        case SCENE_INSIDE_GANONS_CASTLE:
+        case SCENE_BACK_ALLEY_DAY:
+        case SCENE_BACK_ALLEY_NIGHT:
+        case SCENE_MARKET_DAY:
+        case SCENE_MARKET_NIGHT:
+        case SCENE_MARKET_RUINS:
+            return true;
+        default:
+            return false;
+    }
+}
+
+RandomizerCheckArea AreaFromEntranceGroup[] = {
+    RCAREA_INVALID,
+    RCAREA_KOKIRI_FOREST,
+    RCAREA_LOST_WOODS,
+    RCAREA_SACRED_FOREST_MEADOW,
+    RCAREA_KAKARIKO_VILLAGE,
+    RCAREA_GRAVEYARD,
+    RCAREA_DEATH_MOUNTAIN_TRAIL,
+    RCAREA_DEATH_MOUNTAIN_CRATER,
+    RCAREA_GORON_CITY,
+    RCAREA_ZORAS_RIVER,
+    RCAREA_ZORAS_DOMAIN,
+    RCAREA_ZORAS_FOUNTAIN,
+    RCAREA_HYRULE_FIELD,
+    RCAREA_LON_LON_RANCH,
+    RCAREA_LAKE_HYLIA,
+    RCAREA_GERUDO_VALLEY,
+    RCAREA_WASTELAND,
+    RCAREA_MARKET,
+    RCAREA_HYRULE_CASTLE,
+};
+
+RandomizerCheckArea GetCheckArea() {
+    auto scene = static_cast<SceneID>(gPlayState->sceneNum);
+    bool grottoScene = (scene == SCENE_GROTTOS || scene == SCENE_FAIRYS_FOUNTAIN);
+    const EntranceData* ent = GetEntranceData(grottoScene ? ENTRANCE_RANDO_GROTTO_EXIT_START + GetCurrentGrottoId() : gSaveContext.entranceIndex);
+    RandomizerCheckArea area = RCAREA_INVALID;
+    if (ent != nullptr && !IsAreaScene(scene) && ent->type != ENTRANCE_TYPE_DUNGEON) {
+        if (ent->source == "Desert Colossus" || ent->destination == "Desert Colossus") {
+            area = RCAREA_DESERT_COLOSSUS;
+        } else if (ent->source == "Gerudo Fortress" || ent->destination == "Gerudo Fortress") {
+            area = RCAREA_GERUDO_FORTRESS;
+        } else {
+            area = AreaFromEntranceGroup[ent->dstGroup];
+        }
+    }
+    if (area == RCAREA_INVALID) {
+        if (grottoScene && (GetCurrentGrottoId() == -1) && (OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHUFFLE_GROTTO_ENTRANCES) == RO_GENERIC_OFF)) {
+            area = previousArea;
+        } else {
+            area = RandomizerCheckObjects::GetRCAreaBySceneID(scene);
+        }
+    }
+    return area;
+}
+
+bool vector_contains_scene(std::vector<SceneID> vec, const int16_t scene) {
+    return std::any_of(vec.begin(), vec.end(), [&](const auto& x) { return x == scene; });
+}
+
+std::vector<SceneID> skipScenes = {SCENE_GANON_BOSS, SCENE_GANONS_TOWER_COLLAPSE_EXTERIOR, SCENE_GANON_BOSS, SCENE_INSIDE_GANONS_CASTLE_COLLAPSE, SCENE_GANONS_TOWER_COLLAPSE_INTERIOR};
+
+bool EvaluateCheck(RandomizerCheckObject rco) {
+    if (HasItemBeenCollected(rco.rc) && gSaveContext.checkTrackerData[rco.rc].status != RCSHOW_COLLECTED &&
+            gSaveContext.checkTrackerData[rco.rc].status != RCSHOW_SAVED) {
+        SetCheckCollected(rco.rc);
+        return true;
+    }
+    return false;
+}
+
+std::map<RandomizerCheck, QuestItem> QuestItemFromCheck = {
```

Comment:
I don't see this used anywhere

Reply:
You're right, and at this point, I can't remember what I was going to try to use it for.